lib/rules/no-return-value-from-each-iteratee.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileoverview Rule to check the iteratee in a call to _.each does not return a value
+ */
+'use strict';
+
+var _ = require('lodash');
+var lodashUtil = require('../util/lodashUtil');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    return {
+        ReturnStatement: 
+        function (node) {
+            function getSurroundingFunction(astNode) {
+                var ancestors = context.getAncestors();
+                var current = astNode;
+                while (!_.isUndefined(current)) {
+                    if (current.type === 'FunctionExpression' ||
+                        current.type === 'FunctionDeclaration') {
+                        break;
+                    }
+                    current = ancestors.pop();
+                }
+                return current;
```

Comment:
I think this will work, but it needs to be _.findLast, because we have to search ```ancestors``` from the right.